controls/SV-251703.rb

```diff
@@ -3,7 +3,9 @@
   desc 'The "sudo" command allows authorized users to run programs (including shells) as other users, system users, and root. The "/etc/sudoers" file is used to configure authorized "sudo" users as well as the programs they are allowed to run. Some configuration options in the "/etc/sudoers" file allow configured users to run programs without re-authenticating. Use of these configuration options makes it easier for one compromised account to be used to compromise other accounts.
 
 It is possible to include other sudoers files from within the sudoers file currently being parsed using the #include and #includedir directives. When sudo reaches this line it will suspend processing of the current file (/etc/sudoers) and switch to the specified file/directory. Once the end of the included file(s) is reached, the rest of /etc/sudoers will be processed. Files that are included may themselves include other files. A hard limit of 128 nested include files is enforced to prevent include file loops.'
-  desc 'check', 'Verify the operating system specifies only the default "include" directory for the /etc/sudoers file with the following command:
+  desc 'check', 'Note: If the "include" and "includedir" directives are not present in the /etc/sudoers file, this requirement is not applicable.
+
```

Comment:
Delta update: Add a condition to set requirement to NA if `include` or `includedir` are not present in the /etc/sudoers file.

Existing logic flaw: regex wasn't returning `#includedir` that was present. And grep for /etc/sudoers.d was a recursive search.

Proposed code:
```rb
if command('grep include /etc/sudoers').stdout.empty?
  impact 0.0
  describe 'This requirement is not applicable as "include" and "includedir" directives are not present in the /etc/sudoers file' do
    skip 'This requirement is not applicable as "include" and "includedir" directives are not present in the /etc/sudoers file'
  end
elsif virtualization.system.eql?('docker') && !command('sudo').exist?
  impact 0.0
  describe 'Control not applicable within a container without sudo enabled' do
    skip 'Control not applicable within a container without sudo enabled'
  end
else
  describe 'Only the default "include" directory for /etc/sudoers file should be specified' do
    subject { command('grep include /etc/sudoers').stdout.strip }
    it { should cmp "#includedir /etc/sudoers.d"}
  end
  describe 'Nested "include" files or directories within /etc/sudoers.d directory should not exist' do
    subject { command('grep -r include /etc/sudoers.d').stdout }
    it { should be_empty }
  end
end
```

Reply:
After discussing with @ejaronne, updated the matcher to match and allowed for several spaces